src/display.js

```diff
@@ -0,0 +1,120 @@
+import {
+  divFactory,
+  buttonFactory,
+  headerFactory,
+  paragraphFactory
+} from "./elementfactory";
+
+const displayWeatherData = (data, unit) => {
+  const cityName = document.getElementById("citytitle");
+  const content1 = document.getElementById("content1");
+  const content2 = document.getElementById("content2");
+  const card = document.getElementsByClassName("card")[0];
+  let temperatureUnit;
+
+  unit === "metric"
+    ? (temperatureUnit = "Celcius")
+    : (temperatureUnit = "Fahrenheit");
+
+  const { name, main, weather } = data;
+
+  const text = `
+     The temperature is ${Math.floor(main.temp)} 
+     ${temperatureUnit} , a humidity of ${Math.floor(main.humidity)} % 
+     and a pressure of ${Math.floor(main.pressure)} hPa`;
+
+  const text2 = document.createElement("p");
+
+  cityName.innerText = name;
+  content1.innerText = text;
+  text2.innerText = weather[0].description;
+  content2.innerText = "";
+
+  const img = document.createElement("img");
+  img.setAttribute(
+    "src",
+    `http://openweathermap.org/img/w/${weather[0].icon}.png`
+  );
+
+  content2.append(text2, img);
+
+  if (!card.classList.contains("active")) {
+    card.classList.add("active");
+  }
+};
+const handleReset = () => {
+  const current = document.getElementsByClassName("active");
+  if (current.length > 0) {
+    current[0].className = current[0].className.replace("active", "");
+  }
+};
+const handleError = () => {
+  const errorCon = document.getElementById("error-con");
+  if (!errorCon.classList.contains("active")) {
+    errorCon.classList.add("active");
+  }
+};
+
+const createErrorContainer = () => {
+  const el = document.createElement("div");
+  el.setAttribute("id", "error-con");
```

Comment:
Same as before. `setAttribute`/`getAttribute` is useful when you want to set/get custom attributes like:
```
el.setAttribute('my-custom-attribute', value);
```
All the standard attributes are available directly on the element object like this:
```
el.id = 'error-con';
```
You can save yourself a lot of typing.